ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -56,7 +58,7 @@ - (instancetype)initWithListener:(id<CKComponentStateListener>)listener
                  rootIdentifier:rootIdentifier
                  componentClass:componentClass
                           state:initialStateCreator ? initialStateCreator() : [componentClass initialState]
-                     controller:newController(componentClass)];
+                     controller:nil];
```

Comment:
I think it may be worth adding a quick comment here explaining that the controller will be created once the component acquires the scope handle.